InnoClinic.OfficesAPI/API/Middleware/ExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Text.Json;
+using API.Controllers;
+using BLL.Exceptions;
+using FluentValidation;
+
+namespace API.Middleware;
+
+public class ExceptionHandlingMiddleware(ILogger<OfficeController> logger) : IMiddleware
+{
+    public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+    {
+        try
+        {
+            await next(context);
+        }
+        catch (Exception e)
+        {
+            logger.LogError(e, "An unhandled exception occurred: {ExceptionMessage}", e.Message);
+            await HandleExceptionAsync(context, e);
+        }
+    }
+
+    private static async Task HandleExceptionAsync(HttpContext context, Exception exception)
+    {
+        context.Response.ContentType = "application/json";
+
+        if (exception is ValidationException validationException)
+        {
+            context.Response.StatusCode = StatusCodes.Status400BadRequest;
+            await context.Response.WriteAsync(JsonSerializer.Serialize(new
+            {
+                error = "Validation failed",
+                details = validationException.Errors.Select(e => new
+                {
+                    e.PropertyName,
+                    e.ErrorMessage
+                })
+            }));
+            return;
+        }
+
+        context.Response.StatusCode = exception switch
+        {
+            OfficeNotFoundException => StatusCodes.Status404NotFound,
+            OfficePhotoException => StatusCodes.Status400BadRequest,
+            InvalidIdException => StatusCodes.Status400BadRequest,
+            _ => StatusCodes.Status500InternalServerError
+        };
```

Comment:
I wonder, would it be better to define related status codes in the exceptions themselves?